Cleanup YAML jobs

dougbu opened the pull request:
- eliminate useless test publication attempts
- hide non-executing jobs in AzDO UI

dougbu committed:
Remove extra aspnetcore-components-e2e build tasks
- remove `isTestingJob` parameters in YAML
  - controls `PublishTestResults@2` tasks that all fail
- add comments in components-e2e and Helix jobs about "missing" `isTestingJob` parameters

nits:
- remove mention of non-existent `Windows_Templates_Test` job
- use `testResultsFormat` input name for consistency (alias of `testRunner`)

dougbu committed:
Skip useless test publications
- add `enablePublishTestResults` parameter to default-build
  - quarantined-pr contains test jobs but all publish their tests explicitly
- do not attempt to publish TRX files

nit: simplify a YAML snippet
- `parameters.buildDirectory` is never empty (has a default)

dougbu committed:
Eval two conditions during template expansion
- remove jobs that will never execute from UI

dougbu commented:
/azp run

dougbu commented on .azure/pipelines/jobs/default-build.yml:
```diff
@@ -130,10 +133,7 @@ jobs:
-      - BuildDirectory: $(System.DefaultWorkingDirectory)
-    - ${{ if ne(parameters.buildDirectory, '') }}:
-      - BuildDirectory: ${{ parameters.buildDirectory }}
+    - BuildDirectory: ${{ parameters.buildDirectory }}
```
`parameters.buildDirectory` has a non-empty default and is never set to empty.

dougbu commented on .azure/pipelines/jobs/default-build.yml:
```diff
@@ -294,12 +295,12 @@ jobs:
         displayName: Upload cores
 
-    - ${{ if and(eq(parameters.isTestingJob, true), ne(parameters.jobName, 'Windows_Templates_Test')) }}:
+    - ${{ if and(eq(parameters.isTestingJob, true), ne(parameters.enablePublishTestResults, false)) }}:
```
This aligns where we execute this step (`Publish js test results`) with where / when Arcade executes its `Publish XUnit Test Results` step.

dougbu commented on .azure/pipelines/quarantined-pr.yml:
```diff
@@ -84,6 +85,7 @@ jobs:
     agentOs: Windows
     timeoutInMinutes: 90
     isTestingJob: true
+    enablePublishTestResults: false
```
Similar to .azure/pipelines/components-e2e-tests.yml, the three non-Helix jobs in this pipeline need the extra `isTestingJob` steps but not the "normal" test publication (they handle publishing quarantined test results in this YAML). Change avoids similar warnings to those in .azure/pipelines/components-e2e-tests.yml.

dougbu replied:
Technically, `isTestingJob`
1. turns on Arcade and JS test publication (unless overridden by `enablePublishTestResults`)
2. changes the Windows image from `Build.Server.Amd64.VS2019.BT.Open` to `Build.Server.Amd64.VS2019.Open`
3. runs InstallProcDump.ps1, StartDumpCollectionForHangingBuilds.ps1, and FinishDumpCollectionForHangingBuilds.ps1 on Windows build agents
4. runs InstallGoogleChrome.ps1 on Windows build agents

I could remove `isTestingJob: true` from all non-Windows jobs where `enablePublishTestResults: false`, leaving only a few `enablePublishTestResults: false` cases. Left this as-is because we may change how testing jobs are handled for Linux or macOS in the future and the split is pretty clean as I left it between (a) build jobs, (b) Helix submission jobs, and (c) "true" testing jobs (where `isTestingJob: true`).

dougbu commented on .azure/pipelines/ci.yml:
```diff
@@ -149,12 +149,13 @@ stages:
+          condition: notin(variables['DotNetFinalVersionKind'], 'release', 'prerelease')
+          displayName: 'Set CI tags'
+        - script: "echo ##vso[build.addbuildtag]release-candidate"
+          condition: in(variables['DotNetFinalVersionKind'], 'release', 'prerelease')
```
@mmitche I can't find a build tagged with `release-candidate` in aspnetcore-ci-official and we set `$(DotNetFinalVersionKind)` in our `msbuild` code. Did we stop using this pipeline variable at some point❔ If yes, does it even matter for release/2.1❔

mmitche replied:
I don't think this even existed in the 2.1 timeframe. For 3.1+ we set this in eng/Versions.props

dougbu replied:
Alright, I removed the `condition`s using that variable and simplified this to just handle the `daily-build` tag, leaving it easy to find rolling (and manual) builds of all branches.

HaoK commented on .azure/pipelines/ci.yml:
```diff
@@ -716,6 +717,7 @@ stages:
         jobDisplayName: 'Tests: Helix x64'
         agentOs: Windows
         timeoutInMinutes: 240
+        # While isTestingJob is technically true, this job needs none of the normal installations nor test results.
```
Maybe worth just renaming isTestingJob isAzdoTestingJob rather than putting this comment in all the helix yml?

dougbu replied:
Done

dougbu committed:
Address feedback
- rename `parameters.isTestingJob` to `isAzDOTestingJob`
- remove `condition`s about `variables.DotNetFinalVersionKind`
  - variable has not been used in a _very_ long time
- document `parameters.isAzDOTestingJob` and `enablePublishTestResults`

dougbu committed:
!fixup! Add comment for eb9e4c9ea4b9 change
- explain how to test site extension changes

dougbu committed:
!fixup! Simplify one more expression
- missed this one in fc286da97383

dougbu commented:
/fyi just waiting to be sure my `rebase` didn't cause problems anywhere. I also added a tiny reaction to my fc286da97383 (which is the commit that caused conflicts here).
